pkg/gardenlet/controller/backupentry/backup_entry_reconciler.go

```diff
@@ -73,98 +73,104 @@ func (r *reconciler) Reconcile(ctx context.Context, request reconcile.Request) (
+			log.Info("Object is gone, stop reconciling")
 			return reconcile.Result{}, nil
 		}
-		r.logger.Infof("[BACKUPENTRY RECONCILE] %s - unable to retrieve object from store: %v", request.NamespacedName, err)
-		return reconcile.Result{}, err
+		return reconcile.Result{}, fmt.Errorf("error retrieving object from store: %w", err)
 	}
 
 	// Remove the operation annotation if its value is not "restore"
 	// If it's "restore", it will be removed at the end of the reconciliation since it's needed
 	// to properly determine that the operation is "restore, and not "reconcile"
-	backupEntryLogger := logger.NewFieldLogger(logger.Logger, "backupentry", kutil.ObjectName(be))
 	if operationType, ok := be.Annotations[v1beta1constants.GardenerOperation]; ok && operationType != v1beta1constants.GardenerOperationRestore {
 		if updateErr := removeGardenerOperationAnnotation(ctx, gardenClient.Client(), be); updateErr != nil {
-			backupEntryLogger.Errorf("Could not remove %q annotation: %+v", v1beta1constants.GardenerOperation, updateErr)
-			return reconcile.Result{}, updateErr
+			return reconcile.Result{}, fmt.Errorf("could not remove %q annotation: %w", v1beta1constants.GardenerOperation, updateErr)
 		}
 	}
 
 	if be.DeletionTimestamp != nil {
-		return r.deleteBackupEntry(ctx, gardenClient, be)
+		return r.deleteBackupEntry(ctx, log, gardenClient, be)
 	}
 
 	if shouldMigrateBackupEntry(be) {
-		return r.migrateBackupEntry(ctx, gardenClient, be)
+		return r.migrateBackupEntry(ctx, log, gardenClient, be)
 	}
 
 	if !IsBackupEntryManagedByThisGardenlet(be, r.config) {
-		backupEntryLogger.Debugf("Skipping because BackupEntry is not managed by this gardenlet in seed %s", *be.Spec.SeedName)
+		log.V(1).Info("Skipping because BackupEntry is not managed by this gardenlet", "seedName", *be.Spec.SeedName)
 		return reconcile.Result{}, nil
 	}
 
 	// When a BackupEntry deletion timestamp is not set we need to create/reconcile the backup entry.
-	return r.reconcileBackupEntry(ctx, gardenClient, be)
+	return r.reconcileBackupEntry(ctx, log, gardenClient, be)
 }
 
-func (r *reconciler) reconcileBackupEntry(ctx context.Context, gardenClient kubernetes.Interface, backupEntry *gardencorev1beta1.BackupEntry) (reconcile.Result, error) {
-	backupEntryLogger := logger.NewFieldLogger(r.logger, "backupentry", kutil.ObjectName(backupEntry))
-
+func (r *reconciler) reconcileBackupEntry(
+	ctx context.Context,
+	log logr.Logger,
+	gardenClient kubernetes.Interface,
+	backupEntry *gardencorev1beta1.BackupEntry,
+) (
+	reconcile.Result,
+	error,
+) {
 	if !controllerutil.ContainsFinalizer(backupEntry, gardencorev1beta1.GardenerName) {
+		log.Info("Adding finalizer")
 		if err := controllerutils.StrategicMergePatchAddFinalizers(ctx, gardenClient.Client(), backupEntry, gardencorev1beta1.GardenerName); err != nil {
-			backupEntryLogger.Errorf("could not add finalizer to BackupEntry: %+v", err)
 			return reconcile.Result{}, err
```

Comment:
```suggestion
			return reconcile.Result{}, fmt.Errorf("could not add finalizer to BackupEntry: %w", err)
```

Reply:
I think we have not done this in https://github.com/gardener/gardener/blob/4f5d4e97cb6a39f9bb47e7a691c01405ff730f76/pkg/gardenlet/controller/backupbucket/backup_bucket_reconciler.go#L101-L103 because there is this `Adding finalizer` log message above this call, hence if an error occurs it will be the next message which makes clear that it is because of trying to add the finalizer. However, I see that this is not consistent across the set of controllers. Should we add it anyways?

Reply:
I see. We can do it as you prefer.
It would be desirable to clean this up for consistency somewhen though. Maybe it's a good idea to do it in a dedicated PR?

Reply:
Yes, I add it to my notes and do it in the last PR which drops `logrus`. Do you want to have these custom error statements as suggested here or remove them everywhere (and rely on `Adding` finalzer` log)?

Reply:
I tend towards expressive error messages. They make it very easy to pinpoint a given error statement.